### PR TITLE
tune up configuration types

### PR DESCRIPTION
**Pull request by ofrobots**
* Get rid of jsdoc types from comments – not needed now that we TS types. Not entirely sure yet how jsdoc documentation generator handles ts, but we shall find out.
* Proper inheritance for Auth options.
* Fix some TODOs

### CODE REVIEW

File: src/debug.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class Debug extends common.Service {
    *
    * @param {object} options - [Configuration object](#/docs)
```
Comment (DominicKramer) [inline src/debug.ts]: I think we should update the type here or take it out.

File: src/index.ts [hunk 2]
```diff
@@ -27,7 +26,6 @@ let debuglet: Debuglet;
  * with Stackdriver Debug.
  *
  * @param {object=} options - Options
```
Comment (DominicKramer) [inline src/index.ts]: I think we should update the type here or not specify a type.

File: src/debug.ts (hunk 1 above)
Comment (ofrobots) [inline src/debug.ts]: done.

File: src/index.ts (hunk 2 above)
Comment (ofrobots) [inline src/index.ts]: done.

File: src/index.ts [hunk 3]
```diff
@@ -26,31 +25,27 @@ let debuglet: Debuglet;
...
-  const agentConfig = (options as any).debug || options;
+  const agentConfig: DebugAgentConfig =
+      (options as StackdriverConfig).debug || (options as DebugAgentConfig);
 
   // forceNewAgent_ is for testing purposes only.
   if (debuglet && !agentConfig.forceNewAgent_) {
     throw new Error('Debug Agent has already been started');
   }
 
   // TODO: Determine how to remove this cast to `AuthOptions`.
```
Comment (DominicKramer) [inline src/index.ts]: It looks like this TODO can be removed.

Comment (ofrobots) [inline src/index.ts]: Good catch.

**Pull request merged**